### PR TITLE
Add a retrieve promise to DataBehavior

### PR DESCRIPTION
**Pull request by mandragorn**
* The goal is to have a promise that resolves/rejects immediately if
the behavior has already retrieved data or waits until the next fetch
completes before resolving or rejecting.  Resolving or rejecting is
based on the success/failure result of the fetch.

### CODE REVIEW

File: modules/Behavior.js [hunk 1]
```diff
@@ -201,15 +201,34 @@
...
       this.stopListening();
       this.off();
-    }
 
+      this.__isDisposed = true;
+    },
+
+    /**
+     * Method to be invoked when dispose is called. By default calling dispose will remove the
+     * view's on's and listenTo's.
```
Comment (kentmw) [inline modules/Behavior.js]: behavior's on's and listenTo's

File: modules/behaviors/DataBehavior.js [hunk 2]
```diff
@@ -860,10 +903,8 @@
      * @private
      */
     _dispose: function() {
+      Behavior.prototype._dispose.apply(this, arguments);
```
Comment (kentmw) [inline modules/behaviors/DataBehavior.js]: There's no need to do the apply here if the parent is just a Torso.Behavior

File: modules/behaviors/DataBehavior.js [hunk 3]
```diff
@@ -1036,6 +1077,7 @@
     dispose: function() {
```
Comment (kentmw) [inline modules/behaviors/DataBehavior.js]: so you have a public dispose exposed on your behavior? the `this.off` and `this.listenTo` are already being performed by the hidden __dispose. Why not just move the `this.privateCollection.dispose()` to the _dispose you already have?

File: modules/Behavior.js (hunk 1 above)
Comment (mandragorn) [reply modules/Behavior.js]: done

File: modules/behaviors/DataBehavior.js (hunk 2 above)
Comment (mandragorn) [reply modules/behaviors/DataBehavior.js]: done

File: modules/behaviors/DataBehavior.js (hunk 3 above)
Comment (mandragorn) [reply modules/behaviors/DataBehavior.js]: this is on the .data object, not on the DataBehavior itself.  The Data object starts on line 880.

**Pull request merged**